File: 136. Single Number/solution_3.java
//O(nlogn) time
//O(1) space
class Solution {
    public int singleNumber(int[] nums) {
        Arrays.sort(nums);
        int i = 0;
        while (true) {
            if (i == nums.length - 1 || nums[i] != nums[i + 1])
                return nums[i];
            else 
                i += 2;
        }
    }
}

File: 392. Is Subsequence/solution_3.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int sl = s.size(), tl = t.size();
        if (s.size() > t.size())
            return false;
        int i = 0, j = 0;
        for (; i < sl; ++ i) {
            while (j < tl && s[i] != t[j])
                j ++;
            if (j == tl)
                break;
            ++ j;
        }
        return i == sl;
    }
};
//time O(sl + tl)  space O(1)